rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -46,6 +46,20 @@ NodeBase::NodeBase(
   associated_with_executor_(false),
   notify_guard_condition_is_valid_(false)
 {
+
+  // Initialize map_of_mutex object if this is the first NodeBase instance
+  if (this->count_of_instances == 0){
```

Comment:
```suggestion
  if (this->count_of_instances == 0) {
```

Reply:
Also, it's not clear to me that this is actually a count.  That is, it is incremented here, but it is never decremented.  This feels more like it is a boolean flag of "has this been initialized yet?".  There is also the secondary problem of what happens if two nodes are created on two different threads and race here.  It might be better to rely on C++ static initialization for this object, though that has some of its own problems.

Reply:
I'll change that to a bool and add appropriate locking around it.